nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c

```diff
@@ -936,20 +939,41 @@ SetDefaultFontPath(char *path)
                 size = 0,
                 bad;
 
-    /* get enough for string, plus values -- use up commas */
 #ifdef NX_TRANS_SOCKET
-    len = strlen(_NXGetFontPath(path)) + 1;
+    start = (char *) _NXGetFontPath(path);
```

Comment:
Good idea, but wrong implementation:
```C
#ifdef NX_TRANS_SOCKET
    path = (char*)_NXGetFontPath(path);
#endif /* NX_TRANS_SOCKET */

    start = path;
```

Otherwise we'll be checking the NX font path against the originally passed value (`path`) and also write it to `temp_path` etc.